sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatViewModel.kt

```diff
@@ -493,6 +499,52 @@ abstract class ChatViewModel<ARGS: NavArgs>(
         return preview
     }
 
+    private suspend fun handlePaidTextMessageContent(message: Message): LayoutState.Bubble.ContainerThird.Message? {
+        var messageLayoutState: LayoutState.Bubble.ContainerThird.Message? = null
+
+        viewModelScope.launch(mainImmediate) {
+            message?.retrievePaidTextAttachmentUrlAndMessageMedia()?.let { urlAndMedia ->
+                urlAndMedia.second?.host?.let { host ->
+                    urlAndMedia.second?.mediaKeyDecrypted?.let { mediaKeyDecrypted ->
+                        memeServerTokenHandler.retrieveAuthenticationToken(host)?.let { token ->
+
+                            val inputStream = memeInputStreamHandler.retrieveMediaInputStream(
+                                urlAndMedia.first,
+                                token,
+                                mediaKeyDecrypted
+                            )
+
+                            val text = inputStream?.bufferedReader().use { it?.readText() }
+
+                            text?.let { nnText ->
+                                messageLayoutState = LayoutState.Bubble.ContainerThird.Message(text = nnText)
+
+                                nnText.toMessageContentDecrypted()?.let { messageContentDecrypted ->
+                                    updatePaidTextMessageContent(
+                                        message.id,
+                                        messageContentDecrypted
+                                    )
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }.join()
+
+        return messageLayoutState
+    }
+
+    private fun updatePaidTextMessageContent(
+        messageId: MessageId,
+        messageContentDecrypted: MessageContentDecrypted
+    ) {
+        messageRepository.updateMessageContentDecrypted(
+            messageId,
+            messageContentDecrypted
+        )
+    }
```

Comment:
no point to this, just call messageRepository.updateMessgaeContentDecrypted directly.